service/controller/resource/instance/create_cluster_upgrade_requirement_check.go

```diff
@@ -18,20 +18,34 @@ func (r *Resource) clusterUpgradeRequirementCheckTransition(ctx context.Context,
 
 	// Check for changes that must not recycle the nodes but just apply the
 	// VMSS deployment.
+	isCreating := r.isClusterCreating(cr)
 	isScaling, err := r.isClusterScaling(ctx, cr)
 	if err != nil {
 		return "", microerror.Mask(err)
 	}
 
-	if isScaling {
-		// When cluster is scaling we skip upgrading master node[s] and
-		// re-creating worker instances.
+	if isCreating || isScaling {
+		// When cluster is creating or scaling we skip upgrading master node[s]
+		// and replacing worker instances.
 		return DeploymentCompleted, nil
 	}
 
 	return MasterInstancesUpgrading, nil
 }
 
+func (r *Resource) isClusterCreating(cr providerv1alpha1.AzureConfig) bool {
+	// When cluster creation is in the beginning, it doesn't necessarily have
+	// any status conditions yet.
+	if len(cr.Status.Cluster.Conditions) == 0 {
```

Comment:
I am making that up here now but it feels like there can be no conditions for different reasons. Would it be good enough to only take the `Creating` status condition into account? Not blocking here. Just throwing thoughts around.

Reply:
I'm not sure if `Conditions` would be empty other than during creation. It seems that state transitions there use `replace` when patching.

Currently the logic is such that if it cannot pull `Nodes` from tenant API, it doesn't apply patches and this results on Azure often with behaviour that cluster is creating for few reconciliation rounds and then directly lands with `Created`.

Node Pool clusters don't use `statusresource` anymore so I was wondering if it makes sense to spend time fixing that? WDYT?

Reply:
What about adding the `Creating` condition to `kubernetesd` as a workaround/fix for the time being and only rely on the proper condition here in the operator`

Reply:
Otherwise just go with this. Your call.

Reply:
That's a good shout. I'll do that. :+1: